### DISCUSSION TITLE
Showcase...

### DISCUSSION THREAD

**Discussion by matiasdelellis** [dune3d/dune3d#99] [General]
This is the base for some light ceiling roses.. 

![imagen](https://github.com/dune3d/dune3d/assets/733715/b259f334-51b5-43d2-b2a5-1183e6503908)

I hope to make a prototype tomorrow on the 3D printer.. 😬 

Just to thank you... I gave freecad a lot of chances and was never able to achieve anything like that. 😄

**Comment by mfdewit** [comment]
Let me showcase my dune3d projects here as well. I can imagine that @carrotIndustries likes to know his tool is actually being used :hammer: :wrench: 

Coming from FreeCad I wondered where all the options were, but less is more applies here as well. With FreeCad I always ran into problems I could not recover from and the model would be a mess. With Dune3d I actually finished my first two projects successful (that includes the 3d-print). 

Here is my tiller lock, which is a device you can mount on the tiller of a sailboat to lock it into place. A rope goes through the hole and the lever (with a spring) locks the rope so the tiller cannot move.
![Screenshot From 2025-02-20 07-41-24](https://github.com/user-attachments/assets/e47e502a-b875-4c12-a845-1b6b217a308e)

Below is a case for the ESP-GPS-logger which measures speed and distance during windsurfing. I've attached an elastic band so I can mount this on my arm while on the water. Haven't been on the water yet this season, but the dry runs were successful.

![Screenshot From 2025-02-20 07-40-54](https://github.com/user-attachments/assets/7edb9345-a34e-4605-b5fc-eb1c8a6f4541)
![Screenshot From 2025-02-20 07-40-59](https://github.com/user-attachments/assets/8a717b38-cbb0-4e9c-8810-88b800cbd1e5)

All in all I'm quite satisfied with dune3d, however there is some functionality that I really missed. It would be great if the bodies could be moved individually in a linear or rotary motion. 
- For the speedlogger case it would have been very convenient if could have moved the sliding part to check if everything fitted. 
- For the tiller lock the lever rotates and presses against a spring loaded piston. In this case I would like to be able to define a pivot point and rotate the lever to see how far it could move. 

I'm aware the bodies are _tied_ together but being able to add a translation to coordinates and rotation would be very useful in my opinion.

**Reply by carrotIndustries** [reply]
Thats some really good-looking parts you made there :eyes: It's always reassuring to see that people can figure out Dune 3D on their own.

> I'm aware the bodies are tied together but being able to add a translation to coordinates and rotation would be very useful in my opinion.

Yeah, had more or less the same problem when designing https://github.com/carrotIndustries/sd-card-reader-ex/#testing-jig.
I tried working around the issue by leaving the base workplane for the moving part underconstrained. This brings the problem that constraints will have to be specified in a way that they don't depend on the position and rotation of the workplane.

One thing that'll probably be confusing for transformed bodies is that constraints will still apply to them as if the body is not transformed.

Some more things to consider:
 - In tools, we have to undo the transformation to get the cursor to be in the right place in 3D space.
 - How to specify the transformation (translation, rotation)

Just specifying translation and euler angle rotation probably won't cut it.
What might work: Specifying two points in a previous group and using their distance vector to specify the body translation.
For a more complex case that also requires rotation, one could use two workplanes and use them to both specify translation and rotation.

What do you think?

**Reply by mfdewit** [reply]
Thanks! 
I have some trouble understanding the details of your proposed solution, you but in general I can follow. Allow me to make some suggestions or provide feedback at least:

> One thing that'll probably be confusing for transformed bodies is that constraints will still apply to them as if the body is not transformed.

During design time of the current sketch the transformations should all be set to zero. (I think this is what you meant in your first bullet point.) 

Regarding the second bullet: it would be very fancy if one could specify the transformations via either numeric input or a slider component. So when the specific group (body) is selected, the group settings panel on the bottom left of the screen shows these inputs/controls. 

I'm probably simplifying things too much, but wouldn't it be just a matter of only translating the root sketch of the group?
So allow + and - values for X, Y and Z and rotation in degrees for a pivot point which is specified in the sketch. The pivot point specifies the rotation axis.

In practice this would mean that for your SD card reader, you could visualize the movement of the jig by changing the rotation using a slider. And if we can define range limits (min and max degrees) this would open the door to designing complex mechanisms. But now I'm getting carried away :rocket: Sorry :smile: 

Let me know if you would like to elaborate further on this subject. I'm keen to help if you would like to add such features in the future.

**Reply by carrotIndustries** [reply]
Thanks for the suggestions. I now have a rough idea of how to implement it:
The feature will be called explode and there will be an explode slider that goes from 0 to 1 that scales the entered values. If the slider is not at 0, the document becomes read-only.

For each body, one can enter a translation vector (or pick from a line), rotation center and angles (either euler angles or axis/angle).

One can then use the slider to gradually pull apart the document.

As an extra it might make sense to implement multiple 'explode schemes' so that users can quickly switch between various sets of translation/rotation settings.

**Reply by jue89** [reply]
Fancy! I don't know when I really need that feature. But I'll play around with it for hours ... that's for sure!

**Reply by mfdewit** [reply]
Great to hear that you have an idea how to implement this. I like your plan very much and I think it strikes a good balance between possibilities and ease of use. I'm looking forward to be able to test this!
